qa/L0_java_sequence_batcher/Simple.java

```diff
@@ -0,0 +1,676 @@
+        }
+    }
+
+    static class InferResponseComplete extends TRITONSERVER_InferenceResponseCompleteFn_t {
+        @Override public void call (
+            TRITONSERVER_InferenceResponse response, int flags, Pointer userp)
+        {
+          if (response != null) {
+            // Send 'response' to the future.
+            futures.get(userp).complete(response);
+          }
+        }
+    }
+
+    static ConcurrentHashMap<Pointer, CompletableFuture<TRITONSERVER_InferenceResponse>> futures = new ConcurrentHashMap<>();
+    static ResponseAlloc responseAlloc = new ResponseAlloc();
+    static ResponseRelease responseRelease = new ResponseRelease();
+    static InferRequestComplete inferRequestComplete = new InferRequestComplete();
+    static InferResponseComplete inferResponseComplete = new InferResponseComplete();
+
+    static TRITONSERVER_Error
+    ParseModelMetadata(
+        JsonObject model_metadata,
+        boolean[] is_torch_model)
+    {
+      String seen_data_type = null;
+      for (JsonElement input_element : model_metadata.get("inputs").getAsJsonArray()) {
+        JsonObject input = input_element.getAsJsonObject();
+        if (!input.get("datatype").getAsString().equals("INT32")) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_UNSUPPORTED,
+              "sequence qa example only supports model with data type INT32");
+        }
+        if (seen_data_type == null) {
+          seen_data_type = input.get("datatype").getAsString();
+        } else if (!seen_data_type.equals(input.get("datatype").getAsString())) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_INVALID_ARG,
+              "the inputs and outputs of sequence model must have the data type");
+        }
+      }
+      for (JsonElement output_element : model_metadata.get("outputs").getAsJsonArray()) {
+        JsonObject output = output_element.getAsJsonObject();
+        if (!output.get("datatype").getAsString().equals("INT32")) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_UNSUPPORTED,
+              "sequence qa example only supports model with data type INT32");
+        } else if (!seen_data_type.equals(output.get("datatype").getAsString())) {
+          return TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_INVALID_ARG,
+              "the inputs and outputs of sequence' model must have the data type");
+        }
+      }
+
+      is_torch_model[0] =
+          model_metadata.get("platform").getAsString().equals("pytorch_libtorch");
+      return null;
+    }
+
+    static void
+    SetSequenceMetadata(TRITONSERVER_InferenceRequest irequest, long correlation_id, boolean sequence_start, boolean sequence_end)
+    throws Exception
+    {
+
+      FAIL_IF_ERR(
+          TRITONSERVER_InferenceRequestSetCorrelationId(
+              irequest, correlation_id), "Unable to set correlation ID");
+      if(sequence_start) {
```

Comment:
I think we shouldn't use `else if` for `sequence_end` because a sequence can be both starting and ending at the same time and we may need to set both of the flags.

Reply:
Thanks for reviewing, great catch. I updated the code to allow for a request to be both start and end.